sbi/simulators/simutils.py

```diff
@@ -152,13 +160,14 @@ def check_for_batch_reinterpretation_extra_d_uniform(prior):
     """
 
     # Note .batch_shape will always work because and is short-circuiting.
-    if isinstance(prior, Uniform) and prior.batch_shape.numel() > 1:
+    if isinstance(prior, (Uniform, Beta)) and prior.batch_shape.numel() > 1:
```

Comment:
This will happen with all of them that interpret extra dimensions as batch dimensions and would have to be checked exclusively on the basis of the batch_shape, which would be then reinterpreted accordingly *in every case*. But we have agreed we will handle this differently (`MultipleIndependent`, or however we want to call it)..

Reply:
yes, this is actually `outdated` as you can see above in the yellow box.

Reply:
Apologies, I must have overlooked the yellow box.